crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -154,24 +196,29 @@ impl<S: States> Deref for State<S> {
 ///     next_game_state.set(GameState::InGame);
 /// }
 /// ```
-#[derive(Resource, Debug)]
+#[derive(Resource, Debug, Default)]
 #[cfg_attr(
     feature = "bevy_reflect",
     derive(bevy_reflect::Reflect),
-    reflect(Resource, Default)
+    reflect(Resource)
 )]
-pub struct NextState<S: States>(pub Option<S>);
-
-impl<S: States> Default for NextState<S> {
-    fn default() -> Self {
-        Self(None)
-    }
+pub enum NextState<S: FreelyMutableState> {
+    /// No transition has been planned
+    #[default]
+    Unchanged,
+    /// There is a transition planned for state `S`
+    Pending(S),
```

Comment:
nit: Let's update the comments to match the new 'Pending' name.

```suggestion
    /// No state transition is pending
    #[default]
    Unchanged,
    /// There is a pending transition for state `S`
    Pending(S),
```